### PR TITLE
SNOW-1625378 Test write timedelta in I/O

### PR DESCRIPTION
**Pull request by sfc-gh-azhan**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1625378 Test write timedelta in I/O

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

Warn user the timedelta type may be lost during writing back to Snowflake.

### CODE REVIEW

File: tests/integ/modin/io/test_to_csv.py [hunk 1]
```diff
@@ -256,3 +256,36 @@ def test_to_csv_unsupported_params_error(sf_stage, session, kwargs):
...
         pd.DataFrame(native_df).to_csv(stage_location, **kwargs)
+
+
+def test_to_csv_timedelta_local():
+    kwargs = {"columns": ["B"]}
+    native_df = native_pd.DataFrame({"A": ["one", "two", None], "B": [1, 2, 3]})
+    native_path, snow_path = get_filepaths(kwargs, "dataframe_local")
+
+    kwargs = kwargs.copy()
+    kwargs.pop("ext", None)
```
Comment (sfc-gh-mvashishtha) [inline tests/integ/modin/io/test_to_csv.py:267]:
```suggestion
    kwargs = kwargs.copy()
    kwargs.pop("ext", None)
```

I don't think we need this

File: tests/integ/modin/io/test_to_csv.py [hunk 2]
```diff
@@ -256,3 +256,36 @@ def test_to_csv_unsupported_params_error(sf_stage, session, kwargs):
     with pytest.raises(NotImplementedError, match=msg):
         # Write csv to snowflake stage.
         pd.DataFrame(native_df).to_csv(stage_location, **kwargs)
+
+
+def test_to_csv_timedelta_local():
+    kwargs = {"columns": ["B"]}
+    native_df = native_pd.DataFrame({"A": ["one", "two", None], "B": [1, 2, 3]})
```
Comment (sfc-gh-mvashishtha) [inline tests/integ/modin/io/test_to_csv.py:263]: should column `B` have a timedelta type?

File: tests/integ/modin/io/test_to_csv.py [hunk 3]
```diff
@@ -256,3 +256,36 @@ def test_to_csv_unsupported_params_error(sf_stage, session, kwargs):
...
+    kwargs.pop("ext", None)
+    # Write csv with native pandas.
+    native_df.to_csv(native_path, **kwargs)
+    # Write csv with snowpark pandas.
+    pd.DataFrame(native_df).to_csv(snow_path, **kwargs)
+
+    assert_file_equal(snow_path, native_path, is_compressed=True)
+
+
+@pytest.mark.parametrize("kwargs, is_compressed", SERIES_TEST_CASES)
```
Comment (sfc-gh-mvashishtha) [inline tests/integ/modin/io/test_to_csv.py:276]:
```suggestion
```

no need to test all parameter combinations?

File: tests/integ/modin/io/test_to_csv.py [hunk 4]
```diff
@@ -256,3 +256,36 @@ def test_to_csv_unsupported_params_error(sf_stage, session, kwargs):
...
+@sql_count_checker(query_count=1)
+def test_timedelta_to_csv_series_local(kwargs, is_compressed):
+    native_series = native_pd.Series(
+        native_pd.timedelta_range("1 day", periods=3), name="A"
+    )
+    native_path, snow_path = get_filepaths(kwargs, "series_local")
+
+    kwargs = kwargs.copy()
+    kwargs.pop("ext", None)
```
Comment (sfc-gh-mvashishtha) [inline tests/integ/modin/io/test_to_csv.py:285]:
```suggestion
```

no need to use `kwargs`?

File: tests/integ/modin/frame/test_to_snowpark.py [hunk 5]
```diff
@@ -202,3 +203,19 @@ def test_to_snowpark_with_none_data_label_raises(native_pandas_df_basic) -> None
...
+        df = pd.DataFrame(
+            {
+                "a": [1, 2, 3],
+                "b": [4, 5, 6],
+                "t": native_pd.timedelta_range("1 days", periods=3),
+            }
+        )
+        sp = df.to_snowpark(index=False)
+        assert sp.dtypes[-1] == ('"t"', "bigint")
+        print(sp.dtypes)
```
Comment (sfc-gh-evandenberg) [inline tests/integ/modin/frame/test_to_snowpark.py:220]: Should we remove print statements?

File: tests/integ/modin/io/test_to_pandas.py [hunk 6]
```diff
@@ -25,3 +29,7 @@ def test_pd_to_pandas():
         check_index_type=False,
         check_dtype=False,
     )
+    assert_series_equal(
+        pd.to_pandas(pd.Series(data["c"])),
+        native_pd.Series(data["c"]),
+    )
```
Comment (sfc-gh-evandenberg) [inline tests/integ/modin/io/test_to_pandas.py:35]: How come check_index_type=False and check_dtype=False are not needed here?

Comment (sfc-gh-azhan) [reply tests/integ/modin/io/test_to_pandas.py:35]: because to_pandas support timedelta well.

File: tests/integ/modin/io/test_to_csv.py (hunk 2 above)
Comment (sfc-gh-azhan) [reply tests/integ/modin/io/test_to_csv.py:263]: oh sorry, this test should be deleted.

**Review by sfc-gh-mvashishtha — APPROVED** [review]

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py [hunk 7]
```diff
@@ -410,6 +410,26 @@ def _raise_not_implemented_error_for_timedelta(self) -> None:
...
+                    + self._modin_frame.cached_index_column_snowpark_pandas_types
+                )
+                if t is not None
+            ]
+            if snowpark_pandas_types:
+                WarningMessage.lost_type_warning(
+                    method,
+                    ", ".join(snowpark_pandas_types),
+                )
+        except Exception:  # pragma: no cover
```
[the remark this answers is not in the document]
Comment (sfc-gh-azhan) [reply src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:430]: just make sure if something goes wrong, it won't affect the main logic.

[the remark this answers is not in the document]
Comment (sfc-gh-azhan) [reply src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:430]: Actually, we probably don't need this. Please all scenario of the usage have covered by tests. Let me remove it.

**Pull request merged**